keps/sig-node/NNNN-cri-pod-container-stats/README.md

```diff
@@ -0,0 +1,390 @@
+
+| Top level object         | Field                | Currently provided by: | Proposed to be provided by: | 
+|--------------------------|----------------------|------------------------|---------------------| 
+| InterfaceStats (Network) | RxBytes              | cAdvisor               | CRI                 | 
+|                          | RxErrors             | cAdvisor               | CRI                 | 
+|                          | TxBytes              | cAdvisor               | CRI                 | 
+|                          | TxErrors             | cAdvisor               | CRI                 | 
+| CPUStats                 | UsageNanoCores       | cAdvisor               | CRI                 | 
+|                          | UsageCoreNanoSeconds | CRI                    | CRI                 | 
+| MemoryStats              | AvailableBytes       | cAdvisor               | CRI                 | 
+|                          | UsageBytes           | cAdvisor               | CRI                 | 
+|                          | WorkingSetBytes      | CRI                    | CRI                 | 
+|                          | RSSBytes             | cAdvisor               | CRI                 | 
+|                          | PageFaults           | cAdvisor               | CRI                 | 
+|                          | MajorPageFaults      | cAdvisor               | CRI                 | 
+| AcceleratorStats         | Make                 | cAdvisor               | cAdvisor or N/A     | 
+|                          | Model                | cAdvisor               | cAdvisor or N/A     | 
+|                          | ID                   | cAdvisor               | cAdvisor or N/A     | 
+|                          | MemoryTotal          | cAdvisor               | cAdvisor or N/A     | 
+|                          | MemoryUsed           | cAdvisor               | cAdvisor or N/A     | 
+|                          | DutyCycle            | cAdvisor               | cAdvisor or N/A     | 
+| VolumeStats              | All Fields           | Kubelet                | Kubelet             | 
+| Ephemeral Storage        | All Fields           | Kubelet                | Kubelet             | 
+| Rootfs.FsStats           | AvailableBytes       | cAdvisor or N/A        | cAdvisor or N/A     | 
+|                          | CapacityBytes        | cAdvisor or N/A        | cAdvisor or N/A     | 
+|                          | UsedBytes            | CRI                    | CRI                 | 
+|                          | InodesFree           | cAdvisor or N/A        | cAdvisor or N/A     | 
+|                          | Inodes               | cAdvisor or N/A        | cAdvisor or N/A     | 
+|                          | InodesUsed           | CRI                    | CRI                 | 
+| UserDefinedMetrics       | All Fields           | cAdvisor               | N/A                 | 
```

Comment:
Are we proposing to remove them completely?

Reply:
that depends on how we introduce these changes. The proposition as defined is that we don't implement them in CRI API. I believe they're deemed less useful because users can register custom metrics with prometheus in a more idiomatic way--and that UserDefinedMetrics were for a prometheus-less world. 

If these changes are absorbed into the Kubelet by a new stats provider (leaving the old cri stats provider in tact), then the new one would just not support user defined metrics
if these changes were absorbed into the Kubelet by adapting the old cri stats provider, we likely would have to scrape these metrics from cAdvisor.

I'm generally more in favor of the former option. I believe there's a TODO below that notes that we should decide how to handle it

Reply:
I've updated the field to be `cAdvisor or N/A` to better reflect that intention